contents/staticAnalysis/ESLint.md

```diff
@@ -0,0 +1,65 @@
+# ESLint
+
+Author: [Nicholas Chua](https://github.com/nus-cs3281/2018/blob/master/students/nicholasChua/nicholasChua-Resume.md)
+
+## Overview
+
+[ESLint](https://eslint.org/docs/about/) is an open source JavaScript linter.
+
+Code linting is a type of static analysis that helps find problematic patterns or code that does not follow a certain style guideline. JavaScript, being a loosely-typed, interpreted language, is prone to developer errors and therefore linting is especially important when working with JavaScript.
+
+ESLint is shipped with a wide range of [built-in rules](https://eslint.org/docs/rules/) that can be configured on installation. It also offers flexibility by allowing importing your own custom rules via [plugins](https://eslint.org/docs/developer-guide/working-with-plugins). ESLint is widely used by many organizations and has [editor integrations](https://eslint.org/docs/user-guide/integrations#editors) such as error highlighting in many popular editors like Atom and WebStorm.
+
+## How it is used
+
+After [installing with npm](https://eslint.org/docs/user-guide/getting-started#installation-and-usage), ESLint can be configured using [comments or a single configuration file](https://eslint.org/docs/user-guide/configuring#configuring-rules). You can learn how to set them up [here](https://eslint.org/docs/user-guide/configuring). ESLint allows you to disable rules for [specific files and directories](https://eslint.org/docs/user-guide/configuring#ignoring-files-and-directories) or for [specific lines](https://eslint.org/docs/user-guide/configuring#disabling-rules-with-inline-comments).

```

Comment:
>ESLint can be configured using [comments or a single configuration file](https://eslint.org/docs/user-guide/configuring#configuring-rules)

ESLint can be configured in two ways:
* Configuration file: specify all rules in a single file
  {give example here}
* Configuration comments: specify comments in the JS files using comments in a specific format understood by ESLint
  {give example here}